### PR TITLE
Update Rest API and splinter CLI for node public key additions

### PR DESCRIPTION
**Pull request by agunde406**
Updates the Rest API to return public keys with proposals and the full circuit node in the member list of circuits. If challenge authorization is not enabled, the public key is not included. 
```
"members": [
                {
                    "node_id": "n20959",
                    "endpoints": [
                        "tcp://127.0.0.1:18044"
                    ],
                    "public_key": "03f91f722329b99234be43f962e7ce33bbd4f2e72634a1a68f12ad908ca5693f03"
                },
                {
                    "node_id": "n8198",
                    "endpoints": [
                        "tcp://127.0.0.1:28044"
                    ],
                    "public_key": "02e1634dc54bc6ac5675eb3ecafc0c5a833f8e5169d021afcbb9545c122b22d00a"
                }
            ],
```

This requires that the CLI is updated as well, if the public key is not none, it will be shown in the circuit definition
```
Circuit: cXCyT-phOYD
    Display Name: circuit01
    Circuit Status: Active
    Schema Version: 2
    Management Type: manual

    n20959
        Public Key: 03f91f722329b99234be43f962e7ce33bbd4f2e72634a1a68f12ad908ca5693f03
        Service (scabbard): a000
          admin_keys:
              03f91f722329b99234be43f962e7ce33bbd4f2e72634a1a68f12ad908ca5693f03
              038684ef88607ca0e5175fe31b7d94f65b30dc27ef838845f0496eb9c1126c8c82
          peer_services:
              b000
          version:
              2

    n8198
        Public Key: 02e1634dc54bc6ac5675eb3ecafc0c5a833f8e5169d021afcbb9545c122b22d00a
        Service (scabbard): b000
          admin_keys:
              03f91f722329b99234be43f962e7ce33bbd4f2e72634a1a68f12ad908ca5693f03
              038684ef88607ca0e5175fe31b7d94f65b30dc27ef838845f0496eb9c1126c8c82
          peer_services:
              a000
          version:
              2
```

This PR also updates the `splinter circuit propose` CLI command to take challenge authorization type and adding a option to provide public keys for nodes. These changes are also behind an experimental feature challenge-authorization.

```
splinter circuit propose -h
   --auth-type <authorization_type>
            Authorization type for the circuit [default: trust]  [possible values: trust, challenge]
   --node-public-key <node_public_key>...
            Public key for a node that will be used in challenge authorization (<node_id>::<public_key>)
```

### CODE REVIEW

File: cli/src/action/circuit/mod.rs [hunk 1]
```diff
@@ -308,6 +333,34 @@ fn parse_node_argument(node_argument: &str) -> Result<(String, Vec<String>), Cli
...
+        ));
+    }
+
+    let public_key = iter
+        .next()
+        .ok_or_else(|| CliError::ActionError(format!("Missing public key for node '{}'", node_id)))?
+        .to_string();
+    if public_key.is_empty() {
+        return Err(CliError::ActionError(format!(
+            "No public keydetected for node '{}'",
```
Comment (shannynalayna) [inline cli/src/action/circuit/mod.rs:356]: Missing a space here

File: cli/src/action/circuit/builder.rs [hunk 2]
```diff
@@ -377,20 +392,52 @@ fn is_match(service_id_match: &str, service_id: &str) -> bool {
...
-            ))
-        })?;
+        .with_endpoints(endpoints);
+
+    #[cfg(feature = "challenge-authorization")]
+    {
+        if let Some(public_key) = public_key {
+            node_builder = node_builder.with_public_key(&parse_hex(&public_key)?)
+        }
+    }
```
Comment (shannynalayna) [inline cli/src/action/circuit/builder.rs:411]: If public keys are required, should this error if there isn't a public key provided? I noticed while testing that I could specify the `challenge` auth_type without adding the `--node-public-key` option for either node. That proposed circuit would have the challenge auth type. (And I'm able to create a circuit from that)

Comment (shannynalayna) [reply cli/src/action/circuit/builder.rs:411]: Nevermind, realized there's another PR for this concern

File: cli/src/action/circuit/mod.rs (hunk 1 above)
Comment (agunde406) [reply cli/src/action/circuit/mod.rs]: fixed

**Review by shannynalayna — DISMISSED** [review]

File: libsplinter/src/admin/client/event/ws/actix_web_client.rs [hunk 3]
```diff
@@ -415,10 +417,25 @@ impl From<messages::SplinterNode> for CircuitMembers {
...
+                .map(|public_key| to_hex(&public_key)),
         }
     }
 }
 
+#[cfg(feature = "challenge-authorization")]
+fn to_hex(bytes: &[u8]) -> String {
+    let mut buf = String::new();
+    for b in bytes {
+        write!(&mut buf, "{:02x}", b).expect("Unable to write to string");
```
Comment (vaporos) [inline libsplinter/src/admin/client/event/ws/actix_web_client.rs:433]: panic should not occur within the library

Comment (agunde406) [reply libsplinter/src/admin/client/event/ws/actix_web_client.rs]: Updated to use the implementation in the library at hex.rs.

Comment (agunde406) [reply libsplinter/src/admin/client/event/ws/actix_web_client.rs]: This shoud have been used anyway as it is already pulled into the file.

**Review by vaporos — APPROVED** [review]

**Review by shannynalayna — APPROVED** [review]

**Pull request merged**